packages/main/src/RPA/PDF.py

```diff
@@ -1098,48 +1133,70 @@ def get_value_from_anchor(
+                    possible = self._is_match_on_horizontal(direction, item, regexp)
+                elif direction in ["top", "bottom", "up", "down"]:
                     possible = self._is_match_on_vertical(
                         direction, item, strict, regexp
                     )
-                    if possible:
-                        possibles.append(possible)
-            return self._get_closest_from_possibles(direction, possibles)
+                elif direction == "box":
+                    possible = self._is_match_in_box(item)
+                if possible:
+                    possibles.append(possible)
+            if only_closest:
+                return self._get_closest_from_possibles(direction, possibles)
+            else:
+                return possibles
         self.logger.info("NO ANCHOR")
-        return None
+        return possibles
+
+    def _is_within_tolerance(self, base, target):
+        min_target = (
+            target
+            if (target - self.PIXEL_TOLERANCE) < 0
+            else target - self.PIXEL_TOLERANCE
+        )
+        max_target = target + self.PIXEL_TOLERANCE
+        return min_target <= base <= max_target
 
     def _is_match_on_horizontal(self, direction, item, regexp):
         (left, _, right, top) = self.anchor_element.bbox
-        text = None
-        if direction == "right" and item.top == top and item.left > right:
-            self.logger.debug("MATCH %s %s %s", item.boxid, item.text, item.bbox)
-            text = item
-        elif direction == "left" and item.top == top and item.right < left:
-            self.logger.debug("MATCH %s %s %s", item.boxid, item.text, item.bbox)
-            text = item
-        if regexp and text and re.match(regexp, text):
-            return item
-        elif regexp is None and text:
-            return item
-        return None
+        match = False
+        direction_ok = False
+        if (
+            direction == "right"
+            and self._is_within_tolerance(item.top, top)
+            and item.left >= right
+        ):
+            direction_ok = True
+        elif (
+            direction == "left"
+            and self._is_within_tolerance(item.top, top)
+            and item.right < left
+        ):
+            direction_ok = True
+        if regexp and direction_ok and item and re.match(regexp, item.text):
+            match = True
+        elif regexp is None and direction_ok and item:
+            match = True
+        return item if match else None
 
     def _is_match_on_vertical(self, direction, item, strict, regexp):
         (left, bottom, right, top) = self.anchor_element.bbox
         text = None
-        if (direction == "bottom" and item.top < bottom) or (
-            direction == "top" and item.bottom > top
+        direction_down = direction in ["bottom", "down"]
+        direction_up = direction in ["top", "up"]
+        if (direction_down and item.top < bottom) or (
```

Comment:
and should these also be inclusive like the horizontal comparisons? not sure if technically the elements can even overlap.